Collecting React rendering errors (#19470)
DataDog/documentation: content/en/real_user_monitoring/browser/collecting_browser_errors.md

);
{{< /code-block >}}

> **Note**: The [Error Tracking][4] feature processes errors sent with source set to `custom` or `source` and that contain a stack trace. Errors sent with any other source (such as `console`) are not be processed by Error Tracking.

Reviewer: ```suggestion
**Note**: The [Error Tracking][4] feature processes errors sent with source set to `custom` or `source` and that contain a stack trace. Errors sent with any other source (such as `console`) are not processed by Error Tracking.
```

{{% /tab %}}
{{< /tabs >}}

> ### Collect React rendering errors

Reviewer 2: I think there was a mistake in the commits. The title is back to `Collect React rendering errors`

> Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved, which ease developer identifying and fixing errors.

Reviewer: ```suggestion
Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved. The additional information helps developers identify and fix errors.
```

> Error boundaries are React components that catch rendering errors anywhere in its child component tree and display a fallback UI. React error boundaries offer the capability to collect the component stack. Compared to error stack traces, React error boundaries provide more detailed information about the hierarchy of components involved. The additional information helps developers identify and fix errors.
> 
> If your React application throws an error during rendering, React removes its UI from the screen by default. To modify this behavior, you can instrument the React router to monitor React rendering errors with the RUM Browser SDK `addError()` API.
> 
> The collected rendering error stack contains the component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.

Reviewer 2: People looking for a React integration are likely already familiar with Error boundaries. So IMO, there is no need to describe the behaviour of React Error boundaries. 
We could stay concise like for [React router integration](https://docs.datadoghq.com/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#react-router-instrumentation) and only say:

```suggestion
You can instrument the React error boundaries to monitor React rendering errors using the RUM Browser SDK addError() API.
The collected rendering error stack contains the component stack, which is unminified like any other error stacktraces once you’ve uploaded sourcemaps.

To instrument React error boundaries for RUM monitoring, use the following:
```

> **Note**: These instructions are specific to the **React Router v6** library.

Reviewer 2: There is no relation between React router and React Error boundaries
```suggestion
```

Reviewer 2: I think there was a mistake in the commits. The line is back

{{% /tab %}}
{{< /tabs >}}

### React error boundaries instrumentation 

> You can instrument the React error boundaries to monitor React rendering errors using the RUM Browser SDK `addError()` API.

Reviewer 3: <!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about adding a link to react error boundaries doc?

Author: Added, thanks for the suggestions!

> The collected rendering errors contain a component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.

Reviewer 3: <!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about adding a link to sourcemaps upload doc?

> The collected rendering errors contain a component stack, which is unminified like any other error stack traces once you've [uploaded sourcemaps][6].

Reviewer: ```suggestion
The collected rendering errors contain a component stack, which is unminified like any other error stack traces after you [upload sourcemaps][6].
```
Vale made me do it. 😄

{{% /tab %}}
{{< /tabs >}}

### Collect React rendering errors

You can instrument the React error boundaries to monitor React rendering errors using the RUM Browser SDK `addError()` API.

> The collected rendering error stack contains the component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.

Reviewer 2: <!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏The collected rendering errors contain a component stack, which is unminified like any other error stack traces once you've uploaded sourcemaps.
